packages/google_sign_in/google_sign_in/ios/Classes/FLTGoogleSignInPlugin.m

```diff
@@ -77,7 +77,7 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
                                                        ofType:@"plist"];
       if (path) {
         NSMutableDictionary *plist = [[NSMutableDictionary alloc] initWithContentsOfFile:path];
-        [GIDSignIn sharedInstance].clientID = plist[kClientIdKey];
+        [GIDSignIn sharedInstance].clientID = [call.arguments objectForKey:@"clientId"] ?: plist[kClientIdKey];
```

Comment:
I think we also need to check `[[call.arguments objectForKey:@"clientId"] isEqualTo:[NSNull null]]`
Maybe refactor it out to a local bool var:
```
BOOL hasDynamicClientId = [[call.arguments objectForKey:@"clientId"] isKindOfClass:NSString.class]

```

Reply:
You mean checking if `clientId` is a `NSString` object? Because ternary operator `?:` will already check for nullability

Reply:
So we need to check != nil and != [NSNull null]
Checking is a `NSString` covers both cases.